### PR TITLE
Updating release notes for Keycloak 23 with some 'core features' impr…

### LINKED ISSUE #23971: Update release notes for core changes
### Description

This task is just not to forget about update release notes
- #23155
- Something about lightweight access tokens
- #23891
- passkeys (Add contributors)
- Improvements for user-profile?
- Possibly others. See https://github.com/keycloak/keycloak/issues?q=is%3Aissue+is%3Aclosed+milestone%3A23.0.0+label%3Arelease-notes

### PR DESCRIPTION
**Pull request by mposolda**
…ovements

closes #23971

### CODE REVIEW

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 1]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
 Keycloak has preview support for https://fidoalliance.org/passkeys/[Passkeys].
 
 Passkey registration and authentication are realized by the features of WebAuthn.
-Therefore, users of Keycloak can do passkey registration and authentication by existing WebAuthn registraton and authentication.
+Therefore, users of Keycloak can do passkey registration and authentication by existing WebAuthn registration and authentication.
 
 Both synced passkeys and device-bound passkeys can be used for both Same-Device and Cross-Device Authentication.
 However, passkeys operations success depends on the user's environment. Make sure which operations can succeed in https://passkeys.dev/device-support/[the environment].
+Thanks to https://github.com/tnorimat[Takashi Norimatsu] for the contribution and Thanks to https://github.com/thomasdarimont[Thomas Darimont] for the help with the
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:26]:
```suggestion
Thanks to https://github.com/tnorimat[Takashi Norimatsu] for the contribution and thanks to https://github.com/thomasdarimont[Thomas Darimont] for the help with the
```

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 2]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
...
+ideas and testing of this feature.
+
+= WebAuthn improvements
+
+There is new field `Extra Origins` added to WebAuthn policy, which allows better interoperability with non-Web platforms (e.g. native mobile applications).
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:31]:
```suggestion
WebAuthn policy now includes a new field: `Extra Origins`.  It provides better interoperability with non-Web platforms (for example, native mobile applications).
```

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 3]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
...
+Thanks to https://github.com/akunzai[Charley Wu] for the contribution.
 
 = RESTEasy Reactive
 
 Keycloak has switched to RESTEasy Reactive. Applications using `quarkus-resteasy-reactive` should still benefit from a better startup time, runtime performance, and memory footprint, even though not using reactive style/semantics. SPI's that depend directly on JAX-RS API should be compatible with this change. SPI's that depend on RESTEasy Classic including `ResteasyClientBuilder` will not be compatible and will require update, this will also be true for other implementation of the JAX-RS API like Jersey.
+
+= More flexibility for introspection endpoint
+
+In previous versions, introspection endpoint returned automatically most of the claims, which were available in the access token. Now there is new
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:40]:
```suggestion
In previous versions, introspection endpoint  automatically returned most claims, which were available in the access token. Now there is new
```

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 4]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
...
+switch `Add to token introspection` on most of protocol mappers. This allows more flexibility as introspection endpoint can return different
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:41]:
```suggestion
switch `Add to token introspection` on most of protocol mappers. This addition allows more flexibility as introspection endpoint can return different
```

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 5]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
...
+claims than access token. This is first step towards "Lightweight access tokens" support as access tokens can omit lots of the claims, which would be still returned
+by the introspection endpoint. When migrating from previous versions, the introspection endpoint should return same claims, which are returned from access token,
+so the behaviour should be effectively the same by default after the migration. Thanks to https://github.com/skabano[Shigeyuki Kabano] for the contribution.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:44]:
```suggestion
so the behavior should be effectively the same by default after the migration. Thanks to https://github.com/skabano[Shigeyuki Kabano] for the contribution.
```

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 6]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
...
+
+= Feature flag for OAuth 2.0 device authorization grant flow
+
+There is a feature flag added for OAuth 2.0 device authorization grant flow, so it is possible to easily disable this feature. Feature is still enabled by default.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:48]:
```suggestion
OAuth 2.0 device authorization grant flow now includes a feature flag, so you can easily disable this feature. This eature is still enabled by default.
```

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 7]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
...
+Thanks to https://github.com/thomasdarimont[Thomas Darimont] for the contribution.
+
+= Group scalability improvements
+
+Performance around searching of groups is improved for the use-cases with big amount of groups and subgroups. There are improvements, which allow
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:53]:
```suggestion
Performance around searching of groups is improved for the use-cases with many groups and subgroups. There are improvements, which allow
```

File: docs/documentation/release_notes/topics/23_0_0.adoc [hunk 8]
```diff
@@ -19,11 +19,43 @@ https://github.com/tnorimat[Takashi Norimatsu] and https://github.com/dteleguin[
...
+paginated lookup of subgroups. Thanks to https://github.com/alice-wondered[Alice] for the contribution.
+
+= User profile improvements
+
+Declarative user profile is still a preview feature in this release, but we are working hard on promoting it to supported feature. Feedback is welcome.
+If you find any issues or have any improvements in mind, it is welcome to create https://github.com/keycloak/keycloak/issues/new/choose[Github issue],
+ideally with label `area/user-profile`.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/23_0_0.adoc:60]:
```suggestion
Declarative user profile is still a preview feature in this release, but we are working hard on promoting it to a supported feature. Feedback is welcome.
If you find any issues or have any improvements in mind,  you are welcome to create https://github.com/keycloak/keycloak/issues/new/choose[Github issue],
ideally with label `area/user-profile`.
```

**Review by andymunro — COMMENTED** [review]
Thanks, @mposolda.  Just a few suggestions.

**Comment by mposolda** [comment]
@andymunro Thanks! I've updated per your suggestions. I fixed minor typo in one of your suggestions (`This eature`, which should be `This feature`) and squashed the commits.

**Review by ahus1 — APPROVED** [review]

**Pull request merged**